foodsearcher/src/components/SearchForm.tsx

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable prefer-const */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import styles from './SearchForm.module.css';
+import * as List from '../types/optionLists';
+import Dropdown from './Dropdown';
+import { makeUrl } from '../tools/urlMaker';
+
+export default function SearchForm() {
+  const [cuisineType, setCuisineType] = useState('');
+  const [diet, setDiet] = useState('');
+  const [mealType, setMealType] = useState('');
+  const [dishType, setDishType] = useState('');
+  const [q, setQ] = useState('');
+  const navigate = useNavigate();
+  let [skip, setSkip] = useState(true);
+  const params = {
+    cuisineType,
+    dishType,
+    diet,
+    mealType,
+    q,
+  };
+  if (!params.q) params.q = 'i'; // necessary query key
+  const urlWithQuery = makeUrl(params);
+  function handleInput(event: React.ChangeEvent<HTMLInputElement>) {
+    setQ(event.target.value);
+  }
+  function handleSubmit(event: React.FormEvent) {
+    setSkip(false);
+    event.preventDefault();
+    navigate('/search', { replace: false, state: { searchPath: urlWithQuery } });
+  }
+  useEffect(() => {
+    setSkip(true);
+  }, [skip]);
+  return (
+    <form
+      className={styles.form}
+      onSubmit={handleSubmit}
+    >
+      <div>
+        <div>
+          <label htmlFor="dishtype">
+            Type key word:
+            <input type="input" value={q} onChange={handleInput} placeholder="Type your recipe here" />
+          </label>
+        </div>
+        Possible filters:
+        <br />
+        <div>
+          <label htmlFor="dishtype">
+            Dish type
+            <Dropdown value={dishType} onValueChange={(value) => setDishType(value)} placeHolder="Cuisine type" options={List.dishTypeList} />

```

Comment:
```suggestion
            <Dropdown value={dishType} onValueChange={setDishType} placeHolder="Cuisine type" options={List.dishTypeList} />

```
как будто на лекции не был)